pubsub/google/cloud/pubsub/client.py

```diff
@@ -236,3 +238,48 @@ def topic(self, name, timestamp_messages=False):
         :returns: Topic created with the current client.
         """
         return Topic(name, client=self, timestamp_messages=timestamp_messages)
+
+    def subscription(self, name, ack_deadline=None, push_endpoint=None,
+                     retain_acked_messages=None,
+                     message_retention_duration=None):
+        """Creates a subscription bound to the current client.
+
+        Example:
+
+        .. literalinclude:: pubsub_snippets.py
+           :start-after: [START client_subscription]
+           :end-before: [END client_subscription]
+           :dedent: 4
+
+        :type name: str
+        :param name: the name of the subscription to be constructed.
+
+        :type ack_deadline: int
+        :param ack_deadline: (Optional) The deadline (in seconds) by which
+                             messages pulledfrom the back-end must be
+                             acknowledged.
+
+        :type push_endpoint: str
+        :param push_endpoint:
+            (Optional) URL to which messages will be pushed by the back-end.
+            If not set, the application must pull messages.
+
+        :type retain_acked_messages: bool
+        :param retain_acked_messages:
+            (Optional) Whether to retain acked messages. If set, acked messages
+            are retained in the subscription's backlog for a duration indicated
+            by ``message_retention_duration``.
+
+        :type message_retention_duration: :class:`datetime.timedelta`
+        :param message_retention_duration:
+            (Optional) Whether to retain acked messages. If set, acked messages
+            are retained in the subscription's backlog for a duration indicated
+            by ``message_retention_duration``. If unset, defaults to 7 days.
+
+        :rtype: :class:`~google.cloud.pubsub.subscription.Subscription`
+        :returns: Subscription created with the current client.
+        """
+        return Subscription(
+            name, ack_deadline=ack_deadline, push_endpoint=push_endpoint,
+            retain_acked_messages=retain_acked_messages,
+            message_retention_duration=message_retention_duration, client=self)
```

Comment:
Well what's wrong with the two flows that would be supported after this PR:

```python
from google.cloud import pubsub
client = pubsub.Client()

topic = client.topic(topic_name)
sub_create = topic.subscription(sub_name0)
sub_consume = client.subscription(sub_name1)
```

We **could** add a `topic` keyword to this `Client.subscription()` factory, or a `topic_name` keyword but they both have "flaws". The `topic` keyword isn't helpful: if a user already has a `Topic` instance they can just use the `Topic.subscription()` factory. The `topic_name` keyword would miss some parts of the `Topic` constructor: a `timestamp_messages` keyword parameter is also allowed.

PS Your use of "client" confused me (since we have a `Client` class).